test/unit/test_events.py

```diff
@@ -22,6 +21,8 @@
 def get_all_subclasses(cls):
     all_subclasses = []
     for subclass in cls.__subclasses__():
+        if subclass in [NodeInfo, AdapterEventBase, TestLevel, DebugLevel, WarnLevel, InfoLevel, ErrorLevel]:
```

Comment:
The reason we didn't need this before is because we filtered out ABCs for checks. I'd prefer to keep that approach since it means we won't have to remember to update this list. We can do that filtering here since I don't think we'll ever need the ABCs from this function.

Reply:
ABC filtering wasn't working with the Event class as a dataclass.

Reply:
Ahh right.